docs/monitoring.md

```diff
@@ -1421,6 +1421,21 @@ Note: applies to the shuffle service
 - shuffle-server.usedDirectMemory
 - shuffle-server.usedHeapMemory
 
+Note: applies to the shuffle service when the server side flag
+`spark.shuffle.push.server.mergedShuffleFileManagerImpl` set as the appropriate
+org.apache.spark.network.shuffle.MergedShuffleFileManager implementation for the Push-Based Shuffle
+
+- blockAppendCollisions - the number of shuffle push blocks collided in shuffle services
+  as another block for the same reduce partition were being written
+- lateBlockPushes - the number of shuffle push blocks that are received in shuffle service
+  after the specific shuffle merge has been finalized
+- blockBytesWritten - the size of the pushed block data written to file in bytes
+- deferredBlockBytes - the size of the current deferred block parts buffered in memory
+- deferredBlocks - the number of the current deferred block parts buffered in memory
+- staleBlockPushes - the number of stale shuffle block push requests
+- ignoredBlockBytes - the size of the pushed block data that are ignored after the shuffle
+  file is finalized or when a request is for a duplicate block
```

Comment:
nit: 
Can we reorder this so that we go from the common metrics to the more corner cases ?
For example, something like:

* blockBytesWritten
* blockAppendCollisions
* lateBlockPushes
* staleBlockPushes
* deferredBlocks
* deferredBlockBytes
* ignoredBlockBytes

(If this is not the right order, please feel free to reorder them)

Reply:
Thanks for the suggestion, I further put the `staleBlockPushes` to the middle of `deferredBlockBytes ` and `ignoredBlockBytes`, as a stale block push appears in the case of indeterminate stage retries, which I feel is less common to deferred block. Let me know if that's not right. Thanks.

Reply:
Also adjust a bit after @zhouyejoe's suggestion, preview of the corresponding section:
![Screen Shot 2023-01-11 at 2 50 26 PM](https://user-images.githubusercontent.com/31781684/211935150-cc71545b-41ad-4a0c-bc3c-901ef2dd8b5f.png)

Reply:
LGTM.

Reply:
Can we add a new line separation from the earlier list (before the `note: the metrics ...`) ?
All metrics following the note are related to push based shuffle - would be good to visually group them together.

Reply:
Sounds good. Current preview:
![Screen Shot 2023-01-11 at 6 31 34 PM](https://user-images.githubusercontent.com/31781684/211961919-efa4426a-df6f-43b7-8e07-32a455b2bb5f.png)